linkerd/app/inbound/src/detect.rs

```diff
@@ -60,10 +60,56 @@ impl<N> Inbound<N> {
     {
         const TLS_PORT_SKIPPED: tls::ConditionalServerTls =
             tls::ConditionalServerTls::None(tls::NoServerTls::PortSkipped);
+        self.push_detect_http(forward.clone())
```

Comment:
Do we need to push detect_http here? would it be better to keep them separate and compose them only in the main stack?

Reply:
We could do that. I erred on the side of making this change more self-contained, and just splitting it out into a private method so we can write new tests in this module. If you'd prefer to have separate detect TLS and detect HTTP public methods, we could do that instead!